compose/ui/ui-test/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt

```diff
@@ -204,6 +210,21 @@ class SkikoComposeUiTest @InternalTestApi constructor(
         }
     }
 
+    private inline fun <R> renderingAtFrameRate(block: () -> R): R {
+        val scope = CoroutineScope(coroutineContext)
+        return try {
+            scope.launch {
+                while (isActive) {
+                    delay(FRAME_DELAY_MILLIS)
+                    render(mainClock.currentTime)
+                }
+            }
+            block()
+        } finally {
+            scope.cancel("Run completed")
+        }
+    }
+
```

Comment:
>  executing delay-ed coroutines

This is not only about `delay`, right? I.e. any suspend on an external operation (network call) is no longer `non-idle`.

> To fix this advance the test time via mainClock manually, as needed.

Could you add an example?

- Tests that relied on `waitForIdle` (whether explicit or implicit) executing `delay`-ed coroutines will no longer work correctly. To fix this advance the test time via `mainClock` manually, as needed:
  `code`


Also, we need to write what people should do if they waited not only for `delay`, but for something else.

Reply:
> This is not only about delay, right? I.e. any suspend on an external operation (network call) is no longer non-idle.

No, it's only about delay. Suspended coroutines are not registered in `FlushCoroutineDispatcher`.

> Could you add an example?

Where should I add it? In this PR?

Reply:
> Where should I add it? In this PR?

In the release notes as I in a example I provided (just add indentation)

Reply:
> Coroutines suspended not through delay are not registered in FlushCoroutineDispatcher

Looked at the code, and indeed, such tasks are not registered until the coroutine is resumed by something else

Reply:
Open it again, as we still need a code example in Release Notes

Reply:
Added a code example, and documented the change in `advanceTimeBy`.

Reply:
Let's change "Test API" to "Multiple Platforms". We have a strict list of categories in https://github.com/JetBrains/compose-multiplatform-core/edit/jb-main/.github/PULL_REQUEST_TEMPLATE.md. And usually don't categorize it by Compose modules

Reply:
Done